src/components/PostsList.tsx

```diff
@@ -87,0 +1,120 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Post } from '../types/Post';
+import classnames from 'classnames';
+import { getPostsByUserId } from '../api/posts';
+import { User } from '../types/User';
+import { Loader } from './Loader';
+
+type Props = {
+  selectedUser: User | null;
+  onPostSelect: (post: Post | null) => void;
+};
+
+export const PostsList: React.FC<Props> = ({ onPostSelect, selectedUser }) => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const handleOpenPost = useCallback(
+    (post: Post) => {
+      if (selectedPost === post) {
+        setSelectedPost(null);
+        onPostSelect(null);
+      } else {
+        setSelectedPost(post);
+        onPostSelect(post);
+      }
+    },
+    [selectedPost, onPostSelect],
+  );
```

Comment:
Your method names could be more descriptive. Consider following [these](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) naming conventions for methods.